library/core/src/iter/traits/iterator.rs

```diff
@@ -4060,6 +4060,40 @@ pub trait Iterator {
     {
         unreachable!("Always specialized");
     }
+
+    /// Returns `true` if the iterator contains a value.
+    ///
+    /// 'contains' is short-circuiting; in other words, it will stop processing
+    /// as soon as the function finds the item in the `Iterator`.
+    ///
+    /// Performance:
+    /// This method checks the whole iterator, which takes O(n) time.
+    /// If the iterator is sorted, or a hash map please use the appropriate method instead.
```

Comment:
```suggestion
    /// This method checks the whole iterator, which is O(n). If the iterator is a sorted
    /// slice, [`binary_search`](slice::binary_search) may be faster. If this is an iterator
    /// on collections that have a `.contains()` or `.contains_key()` method (such as
    /// [`HashMap`] or [`BTreeSet`]), using those methods directly will be faster.
```

We can be more specific about what the appropriate methods are here. No section heading needed, I think rustdoc would have just merged `Performance: ` into the next paragraph anyway.

You can check the rendering with `./x doc library/core --open`